src/txn/txn_ckpt.c

```diff
@@ -261,8 +261,11 @@ __wt_checkpoint_get_handles(WT_SESSION_IMPL *session, const char *cfg[])
 
     btree = S2BT(session);
 
-    /* Skip files that are never involved in a checkpoint. */
-    if (F_ISSET(btree, WT_BTREE_NO_CHECKPOINT))
+    /*
+     * Skip files that are never involved in a checkpoint. Skip the look aside file as it is,
+     * checkpointed manually later.
+     */
+    if (F_ISSET(btree, WT_BTREE_NO_CHECKPOINT) || F_ISSET(btree, WT_BTREE_LOOKASIDE))
```

Comment:
I think it's odd we skip the metadata file in `__wt_conn_btree_apply()`, but skip the lookaside file here, it seems like handling them both in the same place makes more sense. I wouldn't fight it if that makes the change significantly more difficult.

Reply:
__wt_conn_btree_apply() is called in a few more scenarios than checkpoint, which I think we want to include the LAS file in, e.g. __txn_rollback_to_stable_btree. Additionally I think if I included it there then it would make sense to skip it in __wt_meta_apply_all as well, that change is actually quite ugly so I'm a fan of skipping it separately. As such I'll go ahead and merge.